worker/src/global.rs

```diff
@@ -1,13 +1,25 @@
+use crate::Request as workerRequest;
+use crate::Response as workerResponse;
 use edgeworker_sys::{Response as EdgeResponse, WorkerGlobalScope};
-use js_sys;
-
 use wasm_bindgen::{JsCast, JsValue};
 use wasm_bindgen_futures::JsFuture;
 
-pub async fn fetch_with_str(url: &str) -> Result<EdgeResponse, JsValue> {
+pub async fn fetch_with_str(url: &str) -> Result<workerResponse, JsValue> {
     let worker: WorkerGlobalScope = js_sys::global().unchecked_into();
+
     let promise = worker.fetch_with_str(url);
     let resp = JsFuture::from(promise).await?;
     let resp: EdgeResponse = resp.dyn_into().unwrap();
-    Ok(resp)
+    let worker_resp = workerResponse::from(resp);
+    Ok(worker_resp)
+}
+
+pub async fn fetch_with_request(request: &workerRequest) -> Result<workerResponse, JsValue> {
```

Comment:
Both this and `fetch_with_str` should translate the `JsValue` error to a `Error::JsError`.

Reply:
done